### DISCUSSION TITLE
feature: flash works together with flit and hop.

### DISCUSSION THREAD

**Discussion by meicale** [LazyVim/LazyVim#2387] [Q&A] [answered]
### Did you check the docs?

- [X] I have read all the LazyVim docs

### Is your feature request related to a problem? Please describe.

I prefer flash.nvim to leap.nvim to search for word, and I like the flit.nvim which is based on leap.nvim. How can make this to plugin works together and take better advantage. I install the leap.nvim and disable this keys manully instead of using the lazyvim extra cofnig. But the flit.nvim doesn't work. If I use the lazyvim extra config, the flit.nvim can work but flahs.nvim can not work. Sounds wired, can you solve this?

### Describe the solution you'd like

 I think it is better to make flit.nvim and flash.nvim works together.

### Describe alternatives you've considered

Flash with single letter can works similiar with flit.nvim

### Additional context

Thank you so much for the wonderful work have done!

**Comment by abeldekat** [comment]
Hello @meicale,

`Leap.nvim` and `flash.nvim` share similar functionality but are different plugins.

Quoting from the docs of [flit](https://github.com/ggandor/flit.nvim):

> f/F/t/T motions on steroids, building on the [Leap](https://github.com/ggandor/leap.nvim) interface.

`Flash` has its own `f/F/t/T` motions and is unaware of the `Leap` interface mentioned above.
`Flash` is authored by Folke, `leap` by ggandor.

I think your request is specific to those plugins, and not to `LazyVim`.

**Reply by meicale** [reply]
Yes, outside lazyvim, I cannot make this thing happen.

**Comment by dpetka2001** [comment, marked as answer]
@meicale 
You could try something like the following
```lua
{
    "ggandor/flit.nvim",
    dependencies = {
      "ggandor/leap.nvim",
    },
    opts = {
      labeled_modes = "nx",
    },
  },
  {
    "flash.nvim",
    -- enabled = false,
    opts = {
      modes = {
        char = {
          keys = {},
        },
      },
    },
  }
```
But questions about user configuration should really be asked in the Discussions instead in my opinion.

**Reply by meicale** [reply]
I think this config may lose all the flash.nvim keys, but it doesn't and it works! Thank you!

**Reply by dpetka2001** [reply]
You can also do instead of `keys ={}` the following `enabled = false`, which will disable the `f, F, t, T` functionality all together (since you'll be using `flit` i guess that's not a problem for you) and keep the rest of flash the same.

**Reply by abeldekat** [reply]
Nice solution @dpetka2001!

**Comment by meicale** [comment]
I also found out that hop.nvim is achieved, so I want to replace my hop.nvim action with flash. I use hop to jump to beginning/ending of line/word. That is much more straightforward to just take an action. Any good suggestions to do that? Thank you in advance!
